We have received a request from a user seeking to ascertain how many MA holders there are for metered dose inhalers and enviromentals inhalers. This is related to proposed wotk on phasing out MDIs due to the enviromental impact. We don't have a list of MAs, that should be sought from the MHRA but a very good approximation is what is in the NHS dm+d. 

In [1]:
##import libraries
import pandas as pd
import numpy as np
from ebmdatalab import bq, maps, charts

In [2]:
##we should be able to get this information from the NHS dmd
sql = '''
SELECT
  *
FROM
  `ebmdatalab.dmd2.vw__available_ampps_with_pricing`
WHERE
  vmp_formroute IN ('powderinhalation.inhalation', ## DPI
    'pressurizedinhalation.inhalation') ##MDI
  AND price != 0 ## this will exclude make believe products
ORDER BY
  price_valid_on DESC
  '''
df_inhalerma = bq.cached_read(sql, csv_path='df_inhalerma.csv', use_cache=True)  # add `use_cache=False` to override

df_inhalerma.head()

vmpp_id            ampp_id    ampp_bnf_code    vmpp_bnf_code  \
0  1229311000001105  34215611000001102  0302000N0BGACBG  0302000N0AABGBG   
1  1229311000001105  34675911000001101  0302000N0BJABBG  0302000N0AABGBG   
2  1282011000001105  34023711000001105  0302000N0BIAABF  0302000N0AABFBF   
3  3176611000001101  35430211000001106  0302000C0BSAABE  0302000C0AABEBE   
4  3175511000001107  35430411000001105  0302000C0BSABBF  0302000C0AABFBF   

                                           ampp_name  \
0  AirFluSal 25micrograms/dose / 250micrograms/do...   
1  Aloflute 25micrograms/dose / 250micrograms/dos...   
2  Sereflo 25micrograms/dose / 125micrograms/dose...   
3  Kelhale 50micrograms/dose inhaler (Cipla EU Lt...   
4  Kelhale 100micrograms/dose inhaler (Cipla EU L...   

                                           vmpp_name  dt_price  price  \
0  Fluticasone 250micrograms/dose / Salmeterol 25...      2932   2495   
1  Fluticasone 250micrograms/dose / Salmeterol 25...      2932   2832   
2  Fluticasone 125micrograms/dose / Salmeterol 25...      2345   1499   
3  Beclometasone 50micrograms/dose inhaler CFC fr...       370    520   
4  Beclometasone 100micrograms/dose inhaler CFC f...       742    520   

  price_valid_on  price_prev           price_basis licensed_route  \
0     2019-01-16      2995.0  NHS Indicative Price     Inhalation   
1     2019-01-03      3885.0  NHS Indicative Price     Inhalation   
2     2019-01-02      1649.0  NHS Indicative Price     Inhalation   
3     2019-01-02       606.0  NHS Indicative Price     Inhalation   
4     2019-01-02      1325.0  NHS Indicative Price     Inhalation   

                      vmp_formroute      supplier          gtin gtin_end_date  \
0  pressurizedinhalation.inhalation    Sandoz Ltd  5.050650e+12           NaN   
1  pressurizedinhalation.inhalation         Mylan  5.016695e+12           NaN   
2  pressurizedinhalation.inhalation  Cipla EU Ltd  5.015201e+12           NaN   
3  pressurizedinhalation.inhalation  Cipla EU Ltd  8.901117e+12           NaN   
4  pressurizedinhalation.inhalation  Cipla EU Ltd  8.901117e+12           NaN   

   possible_wholesaler               category  cat_a_weighting  
0                False  Part VIIIA Category C              NaN  
1                False  Part VIIIA Category C              NaN  
2                False  Part VIIIA Category C              NaN  
3                False  Part VIIIA Category C              NaN  
4                False  Part VIIIA Category C              NaN

In [3]:
df_inhalerma.count()

vmpp_id                115
ampp_id                115
ampp_bnf_code          110
vmpp_bnf_code          115
ampp_name              115
vmpp_name              115
dt_price               115
price                  115
price_valid_on          77
price_prev              77
price_basis            115
licensed_route         115
vmp_formroute          115
supplier               115
gtin                   114
gtin_end_date           14
possible_wholesaler    115
category               115
cat_a_weighting         15
dtype: int64

This indicates that there are 115 MDI/DPI in the NHS dmd

In [4]:
##Here we look for MA holders by counting unique suppliers
df_inhalerma.supplier.unique()



array(['Sandoz Ltd', 'Mylan', 'Cipla EU Ltd', 'Aspire Pharma Ltd',
       'GlaxoSmithKline UK Ltd', 'AstraZeneca UK Ltd',
       'Orion Pharma (UK) Ltd', 'Teva UK Ltd',
       'Novartis Pharmaceuticals UK Ltd', 'Boehringer Ingelheim Ltd',
       'Napp Pharmaceuticals Ltd', 'Sanofi', 'Chiesi Ltd',
       'Merck Sharp & Dohme Ltd', 'Meda Pharmaceuticals Ltd',
       'Glenmark Pharmaceuticals Europe Ltd', 'Kent Pharmaceuticals Ltd',
       'Alissa Healthcare Research Ltd'], dtype=object)

There are 18 MA holders